src/arg.c

```diff
@@ -135,9 +293,17 @@ int dcc_scan_args(char *argv[], char **input_file, char **output_file,
     char *a;
     int ret;
 
+    int ignore_range_min = 0, ignore_range_max = 0;
+    if (getenv("DISTCC_RESOLVE_MARCH_NATIVE") &&
```

Comment:
Add documentation about the new environment variable (`man/distcc.1` § `.SH "ENVIRONMENT VARIABLES"`)?

In the last paragraph of DESCRIPTION, I'd add a sentence about tasks that are run locally (preprocessing, linking, and `-m{arch,tune,cpu}=native` unless `$DISTCC_RESOLVE_MARCH_NATIVE` is set).

In CROSS COMPILING another sentence could be added to document the fact that -march=native is handled by the new environment variable.

You could also give different meanings to the `getenv()` value in order to do either:
- "forward": forward native argument to remote machine (dangerous unless remote has same processor)
- "explicit": resolve actual flags by invocation of local compiler, then attempt remote compilation with resolved flags; this incurs a subprocess call penalty.
- "local": just compile locally
- "fail": refuse to process a command line containing these arguments

I am not sure which default would be the sanest.
The best, for a user who wants speed-up, is for them to fixate the equivalent options.